packages/fast-usdc-contract/src/exos/settler.ts

```diff
@@ -159,6 +165,18 @@ export const prepareSettler = (
   },
 ) => {
   assertAllDefined({ statusManager });
+
+  let currentChainRefCached: string;
+  const getCurrentChainRef = (): string => {
+    if (!currentChainRefCached) {
+      if (!baggage.has(CURR_CHAIN_REFERENCE)) {
+        throw Fail`${CURR_CHAIN_REFERENCE} not found in baggage`;
+      }
+      currentChainRefCached = baggage.get(CURR_CHAIN_REFERENCE);
+    }
+    return currentChainRefCached;
+  };
```

Comment:
why does reading this from baggage need to wait until `forward`?

```suggestion
const currentChainRef = baggage.get(CURR_CHAIN_REFERENCE);
```

The `Fail` message is pretty much what `get` will throw if the value isn't there.

Reply:
We don't set `CURR_CHAIN_REFERENCE` in baggage until after the synchronous prelude (since it's not available until then).

This means it will be undefined during `prepareSettler`, and this suggestion won't work. We don't need to wait for `.forward()` to be called, but do we need to wait for `makeSettler` to be called.